lxd/device/device_instance_id.go

```diff
@@ -1,8 +1,14 @@
 package device
 
+import (
+	"github.com/lxc/lxd/lxd/device/config"
+)
+
 // InstanceIdentifier is an interface that allows us to identify an Instance and its properties.
```

Comment:
I wonder if it's worth documenting that this is meant to be a subset of the instance/container interface, entirely comprised of functions which cannot be blocking dependent on when they're called in the lifecycle.

Just to avoid future issues where someone may be tempted to add `IsRunning` or the like to it :)